modules/core/src/main/java/org/apache/ignite/internal/processors/cache/query/continuous/CacheContinuousQueryEventBuffer.java

```diff
@@ -154,6 +185,7 @@ private Object process0(long cntr, CacheContinuousQueryEntry entry, boolean back
         Object res = null;
 
         for (;;) {
+            // Set batch only if batch is null (first attempt).
             batch = initBatch(entry.topologyVersion(), backup);
 
             if (batch == null || cntr < batch.startCntr) {
```

Comment:
We can skip `backupQ.add()` if we already have acked counter more then current counter.

Reply:
I'll fix it under - https://issues.apache.org/jira/browse/IGNITE-13126